rasa_core/dispatcher.py

```diff
@@ -37,12 +37,25 @@ def __init__(self, sender, output_channel, domain):
         self.domain = domain
         self.send_messages = []
 
-    def utter_message(self, message):
-        # type: (Text) -> None
+    def utter_response(self, message):
+        # type: (Dict[Text, Any]) -> None
         """Send a message to the client."""
 
+        if message.get("buttons"):
+            self.utter_button_message(message.get("text"),
+                                      message.get("buttons"))
+        elif message.get("image"):
+            self.utter_message(message.get("text"))
+            self.utter_attachment(message.get("image"))
+        else:
```

Comment:
what if the user specifies both buttons and an image?

Reply:
very good point :+1: